File: golang/2073.time-needed-to-buy-tickets.go
func timeRequiredToBuy(tickets []int, k int) int {
	t := 0
	top := tickets[k]

	for i := 0; i < len(tickets); i++ {
		if i <= k {
			t += min(top, tickets[i])
		} else {
			t += min(top-1, tickets[i])
		}
	}

	return t
}

File: cpp/2073.time-needed-to-buy-tickets.cpp
#include <vector>

using namespace std;

class Solution {
public:
  int timeRequiredToBuy(vector<int> &tickets, int k) {
    int t = 0;
    int top = tickets[k];

    for (int i = 0; i < tickets.size(); ++i) {
      if (i <= k) {
        t += min(top, tickets[i]);
      } else {
        t += min(top - 1, tickets[i]);
      }
    }

    return t;
  }
};